docs: new accessibility page (#1418)
asyncapi/website: pages/docs/community/styleguide/accessibility.md

> ---

Reviewer: Your "Accessibility" Style Guide document provides a solid basis for creating inclusive and accessible content! Nice work, per usual, @thulieblack! Here are a few suggestions to enhance clarity and usefulness.

---- 

## Language
- Include some examples of complex language or technical jargon that should be avoided and how they could be simplified.

## Text
- The point about defining acronyms or abbreviations is crucial. **However, adding a reminder to define these terms upon their first use in any document might be beneficial.**
- "Structure your text in a uniform format" - this point could be expanded to **clarify what a uniform format means in the context of AsyncAPI documentation**.

## Links
- Including guidelines on when and where to use internal vs external links would be helpful. (optional)

## Multimedia
- Could you provide examples of good and bad alt text for images?
- Add more guidance on when it's necessary to use images. For example, complex technical concepts might be better conveyed with diagrams.

## User Interface 
- Clarify what is meant by "correct terminologies for UI elements."
- Add detail on how tables should be formatted.
- The tip about using aria-label attribute when unsure of the icon name is great, but **consider explaining the aria-label attribute for those unfamiliar with accessibility practices.**

## Color usage (missing)
**Please add a section on accessible color usage, especially for contributors who might be creating diagrams or other visual content.**

Author: Added more content on the missing sections. How is it now? @alequetzalli

Reviewer: almost there! just a few fixes left

title: Accessibility
description: This style guide explains how to include accessibility in the documentation.
weight: 20
---

> # Accessibility

Reviewer: ```suggestion
## Accessibility
```

At AsyncAPI, we strive to make our documentation/content inclusive, accessible, and unbiased to everyone. We encourage all contributors to have diversity and inclusivity in mind when writing. To ensure this, we have provided an overview of general guidelines to follow.

> ## Language

Reviewer: ```suggestion
### Language
```

- Be clear and concise when writing. Avoid the use of complex language, technical jargon, and verbose explanations.
- Keep paragraphs and sentences short, simple, and to the point.
- Always maintain a uniform structure. Use descriptive headings and subheadings to make navigation easy.
- Use inclusive language and always keep the reader in mind when writing. 

> ## Text

Reviewer: ```suggestion
### Text
```

- Use the appropriate heading hierarchy. H1 is used for the main heading while H2 to H6 are used for subsection headings. 
- Properly align text for easy readability.
- Avoid using camel case or any unnecessary fonts and formatting.
- Define acronyms or abbreviations and always spell out any signs or symbols.
- Structure your text in a uniform format. 

> ## Links

Reviewer: ```suggestion
### Links
```

- Use descriptive and meaningful link text. For example, do not use phrases like *click here* or *follow this link* instead use *see*.

> - Always use an external link icon if a link opens up in a new tab.

Reviewer: ```suggestion
- Always use an external link icon if a link opens up in a new tab.
```

Why should we use an external link icon if a link opens up in a new tab? Source and reasoning? Want to make sure this is advice we actually believe 🤔

- Make sure your links or URLs are valid and redirect to the correct destination.

> ## Multimedia

Reviewer: ```suggestion
### Multimedia
```

- Alt text must be clear and descriptive. Not more than 50 characters.
- Always use text rather than images, unless necessary.

> - Use SVG instead of PNG or JPEG. It retains quality.

Reviewer: > - Use SVG instead of PNG or JPEG. It retains quality.

In theory agreed... but is this actually something we practice in our docs? So far, I have not seen us use SVGs on our website (or docs). I would remove this line if it is incorrect.

- Provide transcripts and captions for video content.
- Make sure your captions can be translated into various languages.
- Avoid auto-playing media, always provide controls.

> ## UI 

Reviewer: ```suggestion
### User Interface (UI)
```

- Use the correct terminologies for UI elements.
- Format tables correctly and keep the text within the grid.

> -  Use basic HTML for button elements.

Reviewer: ```suggestion
-  Use HTML syntax for button elements.
```

1. What is `basic` HTML? 😄 Why is a button HTML element `basic`? This word choice is wrong because it conveys an assumption and judgement on poor HTML 😂✌🏽 
2. Please add a code snippet to illustrate the button HTML element.

- Be clear and concise when writing. Avoid the use of complex language or technical jargon. For example, instead of using "*MQTT Server Binding*," it could be simplified to "*Protocol-specific information for an MQTT server*," or "*WebSockets Operation Binding*" can be explained as "*Protocol-specific information for WebSockets operation*."
- Keep paragraphs and sentences short, simple, and to the point.
- Always maintain a uniform structure. Use descriptive headings and subheadings to make navigation easy.
- Use inclusive language and always keep the reader in mind when writing. 

### Text
- Use the appropriate heading hierarchy. H1 is used for the main heading while H2 to H6 are used for subsection headings. 
- Properly align text for easy readability.
- Avoid using camel case or any unnecessary fonts and formatting.
- Define acronyms or abbreviations upon the first use in your documentation, and always spell out any signs or symbols.
- Structure your text in a uniform format by maintaining consistent structure throughout your docs. Accurately provide clear and concise descriptions of AsyncAPI concepts and elements as referenced throughout the documentation. 

### Links

> - Use descriptive and meaningful link text. For example, do not use phrases like *click here* or *follow this link* instead use *see*.

Reviewer: ```suggestion
- Use descriptive and meaningful link text. For example, do not use phrases like *click here* or *follow this link* instead use *see*.
```

See..what? To make it descriptive, link text must describe **where the link takes them**.

Please write a new example 😄

Author: 🤣🤣 you are schooling me on accessibility, and I'm glad I'm learning

Reviewer: ♥️🔥🔥🔥 it's definitely a mindset change

### Multimedia
- When adding alt text, providing clear and descriptive information is essential. Avoid using generic terms like "*diagram*," "*image,*" or "*code*", and instead provide specific details such as "*Diagram of a server concept*" or "*Code snippet showing CLI usage*".
- Always use text rather than images, unless necessary. For instance, when explaining complex technical concepts, images convey information better. 
- Use WebP instead of PNG, SVG, or JPEG. It retains quality.
- Provide transcripts and captions for video content.
- Make sure your captions can be translated into various languages.
- Avoid auto-playing media; always provide controls.

### User Interface (UI)
- Use the correct terminologies for UI elements. Avoid using phrases like "*click here*" or "*this button*", and instead opt for descriptive labels such as "*next button*" or "*Go back*". This will allow users to understand the functionality of each UI element better.
- Format tables correctly and keep the text within the grid. This involves using table headers and row/column headers. Make sure tables are user-friendly, easy to navigate, and understandable.

> -  Add icons to describe a function. Use the aria-label attribute when unsure of the icon name. This attribute provides a text alternative to elements that lack accurate descriptions.

Reviewer: ```suggestion
-  Add icons to describe a function. Use the aria-label attribute when unsure of the icon name to provide a text alternative to elements that lack accurate descriptions.
```

Do not start sentences with antecedents (`This`) because that makes them undefined.

## Colors
- It is important to choose color contrasts that ensure user-readability.

> - Diagrams and images must always [follow and respect the accessibility rules](https://webaim.org/resources/contrastchecker/).

Reviewer: ```suggestion
- Diagrams and images must always follow and respect [color contrast accessibility rules](https://webaim.org/resources/contrastchecker/).
```
